Rules.modular

```diff
@@ -138,26 +146,25 @@ $(tmpdir)/global_bools.conf: M4PARAM += -D self_contained_policy
 $(tmpdir)/global_bools.conf: $(m4support) $(tmpdir)/generated_definitions.conf $(globalbool) $(globaltun)
 	$(verbose) $(M4) $(M4PARAM) $^ > $@
 
-$(tmpdir)/all_interfaces.conf: $(m4support) $(all_interfaces) $(m4iferror)
-	@test -d $(tmpdir) || mkdir -p $(tmpdir)
+$(tmpdir)/all_interfaces.conf: | $(tmpdir)
+$(tmpdir)/all_interfaces.conf: $(tmpdir)/m4support.m4f $(all_interfaces) $(m4iferror)
 	@cat $(m4divert) > $@
-	$(verbose) $(M4) $^ > $(tmpdir)/$(@F).tmp
-	$(verbose) $(SED) -e s/dollarsstar/\$$\*/g $(tmpdir)/$(@F).tmp >> $@
+	$(verbose) $(M4) $(M4PARAM) -R $^ | $(SED) -e s/dollarsstar/\$$\*/g >> $@
 	@cat $(m4undivert) >> $@
 
+$(tmpdir)/all_te_files.conf: | $(tmpdir)
 $(tmpdir)/all_te_files.conf: M4PARAM += -D self_contained_policy
 $(tmpdir)/all_te_files.conf: $(m4support) $(tmpdir)/generated_definitions.conf $(tmpdir)/all_interfaces.conf $(base_te_files) $(m4terminate)
 ifeq "$(strip $(base_te_files))" ""
 	$(error No enabled modules! $(notdir $(mod_conf)) may need to be generated by using "make conf")
 endif
-	@test -d $(tmpdir) || mkdir -p $(tmpdir)
 	$(verbose) $(M4) $(M4PARAM) -s $^ > $@
 
+$(tmpdir)/post_te_files.conf: | $(tmpdir) $(tmpdir)/all_te_files.conf
```

Comment:
This is a change of the dependency type for `$(tmpdir)/all_te_files.conf` from *check if older* to *check if exists*.